### ISSUE TITLE
Crash when Beeftext replace combo

### ISSUE THREAD

**Issue by Rukhazon** [xmichelo/Beeftext#516]
Hi, since I updated Windows 11 to 22H2 release, Beeftext crash just after delete combo (but did not write the substitute).
There is no error message, the app silently quit.

**Comment by realandsalv** [comment]
I had the same problem. I uninstalled Beeftext, restarted my pc, and reinstalled it (v14.0). It works fine for me now — hope this helps somehow.

**Comment by rafitamolin** [comment]
Crashing is pretty intense, before it happened every now and then, now 3 of 5 tries end up crashing Beeftext.

**Comment by xmichelo** [comment]
Have you tried re-installing Beeftext?

**Comment by Rukhazon** [comment]
> Have you tried re-installing Beeftext?

Yes I do, but same crash happen with both (portable and installable) versions.

**Comment by xtedx** [comment]
Also crashing for me, this is the content of error report log from event viewer, but nothing in the beeftext log. I hope it is useful for debugging?
Have not tried re-installing, it helps a after restart, but i notice it could be after going standby and/or hibernation overnight.

I don't know if it is related, my windows 11 can't sleep, wont go idle, won't activate screen saver even since before 2022h2.
but beeftext crashing only after 2022h2.

[Report.wer.txt](https://github.com/xmichelo/Beeftext/files/9857069/Report.wer.txt)

**Comment by devfacto-chadross** [comment]
I have been having this problem since updating to Windows 11 to 22H2 as well. I've managed to capture some event log details, so I hope they are helpful to resolving this problem.



**Error**: Application Error
```
Faulting application name: Beeftext.exe, version: 14.0.0.0, time stamp: 0x6279edcc
Faulting module name: ntdll.dll, version: 10.0.22621.755, time stamp: 0x8a328c67
Exception code: 0xc0000374
Fault offset: 0x000000000010c249
Faulting process ID: 0x0x8BC
Faulting application start time: 0x0x1D8EE362AA9F46B
Faulting application path: C:\Program Files\Beeftext\Beeftext.exe
Faulting module path: C:\WINDOWS\SYSTEM32\ntdll.dll
Report ID: 58c45c60-e3fd-4d21-974d-ee07ca9a92f2
Faulting package full name: 
Faulting package-relative application ID: 


<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
[-](https://github.com/xmichelo/Beeftext/issues/516#) <System>
  <Provider Name="Application Error" Guid="{a0e9b465-b939-57d7-b27d-95d8e925ff57}" /> 
  <EventID>1000</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>100</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x8000000000000000</Keywords> 
  <TimeCreated SystemTime="2022-11-03T15:17:31.6579996Z" /> 
  <EventRecordID>6021</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="16004" ThreadID="1136" /> 
  <Channel>Application</Channel> 
  <Computer>DFTDL21008CR</Computer> 
  <Security UserID="S-1-12-1-3639927005-1093335723-4170962334-2882190746" /> 
  </System>
[-](https://github.com/xmichelo/Beeftext/issues/516#) <EventData>
  <Data Name="AppName">Beeftext.exe</Data> 
  <Data Name="AppVersion">14.0.0.0</Data> 
  <Data Name="AppTimeStamp">6279edcc</Data> 
  <Data Name="ModuleName">ntdll.dll</Data> 
  <Data Name="ModuleVersion">10.0.22621.755</Data> 
  <Data Name="ModuleTimeStamp">8a328c67</Data> 
  <Data Name="ExceptionCode">c0000374</Data> 
  <Data Name="FaultingOffset">000000000010c249</Data> 
  <Data Name="ProcessId">0x8bc</Data> 
  <Data Name="ProcessCreationTime">0x1d8ee362aa9f46b</Data> 
  <Data Name="AppPath">C:\Program Files\Beeftext\Beeftext.exe</Data> 
  <Data Name="ModulePath">C:\WINDOWS\SYSTEM32\ntdll.dll</Data> 
  <Data Name="IntegratorReportId">58c45c60-e3fd-4d21-974d-ee07ca9a92f2</Data> 
  <Data Name="PackageFullName" /> 
  <Data Name="PackageRelativeAppId" /> 
  </EventData>
  </Event>
```

**Comment by realandsalv** [comment]
Just want to add an update:

After I thought a reinstall fixed the crashing issues, a more recent Windows 11 update also caused Beeftext to constantly crash again... so back to square one. 🙃

**Comment by hubjective** [comment]
Back to report that it's happening to me again, as well. If I had to say what the most common use case that causes the crash is, it would be typing the combo in a Chrome web page input field. So for sites A, B, C, and D, it will work without issue. But on site E, it will crash every time I type the combo in the input field. That is what happened to generate the crash log below. There wasn't anything in the Beeftext log.

Log Name:      Application
Source:        Application Error
Date:          11/10/2022 8:53:01 AM
Event ID:      1000
Task Category: Application Crashing Events
Level:         Error
Keywords:      
User:          
Computer:      
Description:
Faulting application name: Beeftext.exe, version: 14.0.0.0, time stamp: 0x6279edcc
Faulting module name: ntdll.dll, version: 10.0.22621.608, time stamp: 0xf2e8a5ab
Exception code: 0xc0000374
Fault offset: 0x000000000010c1c9
Faulting process id: 0x0x839C
Faulting application start time: 0x0x1D8F392F295BAB7
Faulting application path: C:\Program Files\Beeftext\Beeftext.exe
Faulting module path: C:\WINDOWS\SYSTEM32\ntdll.dll
Report Id: e0305a91-a37a-4b3e-9e35-65e01c0cc022
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" Guid="{a0e9b465-b939-57d7-b27d-95d8e925ff57}" />
    <EventID>1000</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>100</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8000000000000000</Keywords>
    <TimeCreated SystemTime="2022-11-10T13:53:01.8089717Z" />
    <EventRecordID>7069</EventRecordID>
    <Correlation />
    <Execution ProcessID="34604" ThreadID="37580" />
    <Channel>Application</Channel>
    <Computer></Computer>
    <Security UserID="" />
  </System>
  <EventData>
    <Data Name="AppName">Beeftext.exe</Data>
    <Data Name="AppVersion">14.0.0.0</Data>
    <Data Name="AppTimeStamp">6279edcc</Data>
    <Data Name="ModuleName">ntdll.dll</Data>
    <Data Name="ModuleVersion">10.0.22621.608</Data>
    <Data Name="ModuleTimeStamp">f2e8a5ab</Data>
    <Data Name="ExceptionCode">c0000374</Data>
    <Data Name="FaultingOffset">000000000010c1c9</Data>
    <Data Name="ProcessId">0x839c</Data>
    <Data Name="ProcessCreationTime">0x1d8f392f295bab7</Data>
    <Data Name="AppPath">C:\Program Files\Beeftext\Beeftext.exe</Data>
    <Data Name="ModulePath">C:\WINDOWS\SYSTEM32\ntdll.dll</Data>
    <Data Name="IntegratorReportId">e0305a91-a37a-4b3e-9e35-65e01c0cc022</Data>
    <Data Name="PackageFullName">
    </Data>
    <Data Name="PackageRelativeAppId">
    </Data>
  </EventData>
</Event>

**Comment by devfacto-chadross** [comment]
This crash is still happening for me on my Win11 install. It doesn't happen every time I trigger a combo, but it seems to happen mostly on my combos that include `#{clipboard}` in the snippet.

**Comment by andrewtorres** [comment]
Happening to me also when the clipboard is loaded with the content of an Excel cell. If the value is copied from the formula bar instead (plain text format), the combo works and the app doesn't crash.

**Comment by hubjective** [comment]
> 

Coming back to say that I'm seeing the same correlation with Excel contents being on the clipboard. Nice catch @andrewtorres

**Comment by xmichelo** [comment]
Hi.

@andrewtorres You nailed it. It seems to me that the problem is linked to a bitmap being stored in the clipboard. As strange as it may seem, When copying a cell or a set of cells, in addition to the text, Excel also put a a bitmap image of the cells in the clipboard.

I've opened a [specific issue](https://github.com/xmichelo/Beeftext/issues/543). Can you please check the test build in the comment?

**Comment by andrewtorres** [comment]
Hi @xmichelo, yes, the new build is working now for me! That seemed to have done the trick! 
Many thanks!
Andrew
